iroh/src/client.rs

```diff
@@ -420,7 +421,7 @@ where
     /// List all collections.
     pub async fn list_collections(
         &self,
-    ) -> Result<impl Stream<Item = Result<BlobListCollectionsResponse>>> {
+    ) -> Result<impl Stream<Item = Result<RpcResult<BlobListCollectionsResponse>>>> {
```

Comment:
shouldn't we flatten these results? or is there real value nesting these?

Reply:
I am not a big fan of flattening these.

One is a network error, the other is an internal error on the remote side. Sure, you can flatten them and make it an anyhow, but I think they are very different things that you also might want to report in different ways.

Reply:
internally I would agree, but from a publics api perspective it is very confusing, especially given that the outer one is already a `anyhow::Result`

Reply:
if you care, you can always downcast back as a consumer